### PR TITLE
Managed Groups OIDC Repo

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: internal/auth/subtype.go [hunk 1]
```diff
@@ -58,6 +58,18 @@ var (
 	_ Account = (*password.Account)(nil)
 )
 
+type ManagedGroup interface {
```
Comment (jimlambrt) [inline internal/auth/subtype.go:61]: where/how is this used?  I couldn't find any current references.  Perhaps, it's need in future managed groups PRs?

File: internal/auth/oidc/repository_managed_group_test.go [hunk 2]
```diff
@@ -0,0 +1,950 @@
...
+	tests := []struct {
+		name            string
+		in              *ManagedGroup
+		opts            []Option
+		want            *ManagedGroup
+		wantIsErr       errors.Code
+		wantErrMsg      string
+		wantErrContains string
+	}{
+		{
```
Comment (jimlambrt) [inline internal/auth/oidc/repository_managed_group_test.go:54]: mind adding a test for no scope id?

File: internal/auth/oidc/testing.go [hunk 3]
```diff
@@ -176,6 +176,24 @@ func TestAccount(t *testing.T, conn *gorm.DB, am *AuthMethod, subject string, op
 	return a
 }
 
+// TestManagedGroup creates a test oidc managed group.
+func TestManagedGroup(t *testing.T, conn *gorm.DB, am *AuthMethod, filter string, opt ...Option) *ManagedGroup {
```
Comment (jimlambrt) [inline internal/auth/oidc/testing.go:180]: mind adding some tests that would cover different scenarios where ManagedGroup.validate(...) would fail?

**Review by jimlambrt — APPROVED** [review]
Just a couple minor things with testing that you might want to address.

File: internal/auth/subtype.go (hunk 1 above)
Comment (jefferai) [reply internal/auth/subtype.go:61]: Yep, it's used later in the service handler (PR to come).

File: internal/auth/oidc/repository_managed_group_test.go (hunk 2 above)
Comment (jefferai) [reply internal/auth/oidc/repository_managed_group_test.go:54]: Done.

File: internal/auth/oidc/testing.go (hunk 3 above)
Comment (jefferai) [reply internal/auth/oidc/testing.go:180]: Done.

**Review by jimlambrt — APPROVED** [review]

**Review by vancluever — APPROVED** [review]
LGTM here! Being new still this has been mostly an informative review for me, but this has been great to learn about how repositories work!

**Pull request merged**